alphadia/workflow/peptidecentric/optimization_handler.py

```diff
@@ -363,33 +368,62 @@ def _process_batch(self):
             ),
         )
 
-        feature_df, fragment_df = extraction_handler.extract_batch(
-            (self._dia_data, self._dia_data_ng)
-            if self._dia_data_ng is not None
-            else self._dia_data,
+        candidates_df = extraction_handler.select_candidates(
+            self._dia_data,
             self._optlock.batch_library,
         )
-        self._optlock.update_with_extraction(feature_df, fragment_df)
+
+        if self._config["search"]["extraction_backend"] == "classic":
+            features_df, fragments_df = (
+                extraction_handler.score_and_quantify_candidates(
+                    candidates_df,
+                    self._dia_data,
+                    self._optlock.batch_library,
+                )
+            )
+
+            self._optlock.update_with_extraction(features_df, fragments_df)
+
+            decoy_strategy = (
+                "precursor_channel_wise"
+                if self._config["fdr"]["channel_wise_fdr"]
+                else "precursor"
+            )
+
+            precursor_df = self._fdr_manager.fit_predict(
+                self._optlock.features_df,
+                decoy_strategy=decoy_strategy,
+                competetive=self._config["fdr"]["competetive_scoring"],
+                df_fragments=self._optlock.fragments_df,
+                version=self._optimization_manager.classifier_version,
+            )
+        else:
+            features_df = extraction_handler.score_candidates(
+                candidates_df, self._dia_data, self._optlock.batch_library
+            )
+
+            _, fragments_df = extraction_handler.quantify_candidates(
```

Comment:
I think we have a bit of an (sleeping?) issue here. This might actually destroy the precursor fragment coupling trough `frag_strat_start/stop_idx`